ansible/site.yml

```diff
@@ -0,0 +1,45 @@
+- hosts: all
+  sudo: true
+  environment: proxy_env
+  tasks:
+    - hostname: "name={{ hostname }}"
+    - template: src=templates/envvar.sh.j2 dest=/etc/profile.d/envvar.sh owner=root group=root mode=0755
+    - apt: "update_cache=yes name={{ item }}"
+      with_items:
+        - unzip
+        - vim-nox
+        - curl
+        - python-software-properties
+        - git 
+    - shell: creates=/usr/local/go/bin/go curl -L https://storage.googleapis.com/golang/go1.4.2.linux-amd64.tar.gz | tar -xvz -C /usr/local
```

Comment:
It does, note the `creates` attributes on several lines. This is one mechanism to determine if something is already installed, packages and the get_url stuff will manage the rest of the idempotent concerns.

I hope to separate this into ansible "build time" and "run time" concerns so that we can build our own images with ease, but I was hoping this could make it in first and I could span out from there.